runtime/vanilla/src/lib.rs

```diff
@@ -566,17 +566,37 @@ impl pallet_loans::Config for Runtime {
     type UnixTime = Timestamp;
 }
 
-pub type TimeStampedPrice = orml_oracle::TimestampedValue<PriceWithDecimal, Moment>;
 pub struct AggregatedDataProvider;
 impl DataProvider<CurrencyId, TimeStampedPrice> for AggregatedDataProvider {
     fn get(key: &CurrencyId) -> Option<TimeStampedPrice> {
         Oracle::get(key)
     }
 }
+
+impl DataProviderExtended<CurrencyId, TimeStampedPrice> for AggregatedDataProvider {
+    fn get_no_op(key: &CurrencyId) -> Option<TimeStampedPrice> {
+        Oracle::get_no_op(key)
```

Comment:
Shouldn't this is Prices::get_no_op(key)?

Reply:
It's Oracle actually, the method you point out in runtime will be invoked by `T::Source::get_no_op()` in the Price pallet.

Reply:
Ok, I see.